### ISSUE TITLE
WKT2 support

### ISSUE THREAD

**Issue by snowman2** [Toblerity/Fiona#977]
Related to:
- https://github.com/OSGeo/PROJ/issues/2426#issuecomment-727937449
- https://github.com/geopandas/geopandas/issues/1697

Currently error is produced here:
```python
import fiona
schema = {"geometry": "Point", "properties": {"zero": "int"}}
feature = {
    "geometry": {"type": "Point", "coordinates": (0, 0)},
    "properties": {"zero": "0"},
}
target_crs = 'PROJCS["IaRCS_04_Sioux_City-Iowa_Falls_NAD_1983_2011_LCC_US_Feet",GEOGCS["GCS_NAD_1983_2011",DATUM["D_NAD_1983_2011",SPHEROID["GRS_1980",6378137.0,298.257222101]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433]],PROJECTION["Lambert_Conformal_Conic"],PARAMETER["False_Easting",14500000.0],PARAMETER["False_Northing",8600000.0],PARAMETER["Central_Meridian",-94.83333333333333],PARAMETER["Standard_Parallel_1",42.53333333333333],PARAMETER["Standard_Parallel_2",42.53333333333333],PARAMETER["Scale_Factor",1.000045],PARAMETER["Latitude_Of_Origin",42.53333333333333],UNIT["Foot_US",0.3048006096012192]]'
with fiona.open("./test.shp", "w", driver="ESRI Shapefile", schema=schema, crs=target_crs) as collection:
    collection.write(feature)
```

Solution: use WKT2. Need to do some research on how to do that.

**Comment by snowman2** [comment]
Here is a workaround for anyone who might run into this:

Solution: use `OSR_WKT_FORMAT="WKT2_2018"` or  `OSR_WKT_FORMAT="WKT1_ESRI"` with `fiona.Env`:

Write:
```python
with fiona.Env(OSR_WKT_FORMAT="WKT2_2018"), fiona.open("./test.shp", "w", driver="ESRI Shapefile", schema=schema, crs=target_crs) as collection:
    collection.write(feature):
```
Read:
```python
with fiona.Env(OSR_WKT_FORMAT="WKT2_2018"), fiona.open("./test.shp") as collection:
   print(collection.crs)
```

**Comment by mwtoews** [comment]
Closed by #979
